cmd/server/main.go

```diff
@@ -196,6 +187,32 @@ func main() {
 	log.Fatal(http.ListenAndServe(cfg.ListenAddr, router))
 }
 
+func validateNetworkName(cfg *config, cChainClient client.Client) error {
+	if cfg.Mode == service.ModeOffline && cfg.NetworkName == "" {
+		return fmt.Errorf("network name is not provided, can't fetch network name in offline mode")
+	}
+
+	log.Println("fetching network name from rpc...")
+	networkName, err := cChainClient.GetNetworkName(context.Background())
+	if err != nil {
+		return fmt.Errorf("cant fetch network name: %w", err)
+	}
+
+	if cfg.NetworkName == "" {
```

Comment:
Should we use this the other way around, and put it before fetching the network name from the client?

```go
if cfg.NetworkName != "" {
  if cfg.NetworkName != networkName {
    // return error
  }

  return nil
}

// fetch from client
```

Reply:
Indeed I was calling the backend client even in case service was offline. Fixed, not exactly as you suggested, but in an equivalent way